update-repo.py

```diff
@@ -0,0 +1,166 @@
+import subprocess
+import sys
+import os
+
+import shutil
+
+
+"""
+This script streamlines the process of copying a file of any type into each workflows directory, 
+followed by creating a PR into the GitHub feedstock repository.
+
+How to use:
+
+python /path/.../update-repo.py <filepath-you-want-to-copy-in-your-local-machine> <filepath-in-the-github-repo>
+e.g. python /path/.../update-repo.py ../dev/example/test.yml  .github/workflows 
+
+Workflow:
+
+- Walk through all directories to find corresponding filepath
+- Copy file to desired directories
+- Fetch the user's GitHub username using the GitHub CLI for authentication
+- Commit these changes, pushes them to GitHub, and creates a PR
+"""
+
+"""
+Utility Functions
+"""
+
+
+def run_command(command, cwd=None):
+    """Run a shell command in the specified directory."""
+    subprocess.run(
+        command,
+        check=True,
+        shell=True,
+        text=True,
+        cwd=cwd,
+    )
+
+
+"""
+Core Functionalities - sync with the main branch, copy desired file, 
+                       pass a variable to the file, and create PR
+"""
+
+
+def sync_with_main_branch(dirpath):
+    # Check out main and pull the latest changes
+    run_command("git checkout main", cwd=dirpath)
+    run_command("git pull upstream main", cwd=dirpath)
+
+
+def pass_variable(file_path, package_name):
+    # Read the file contents
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    # Look for the line containing 'project:' and update it
+    updated_lines = []
+    for line in lines:
+        if 'project:' in line:
+            updated_lines.append(f'      project: {package_name}\n')
+        else:
+            updated_lines.append(line)
+
+    # Write the updated lines back to the file
+    with open(file_path, 'w') as file:
+        file.writelines(updated_lines)
+
+
+def copy_file(source_file, dirpath, workflow_dir, dest_file, username):
+    # Create and switch to a new branch named after the new version
+    run_command(f"git checkout -b {source_file}", cwd=dirpath)
+
+    # Copy the file
+    shutil.copy2(source_file, dest_file)
+    print(f'File copied to {workflow_dir}')
+
+    # Pass the package name into the file
+    package_name = dirpath.split('/')[1]
+    pass_variable(dest_file, package_name)
```

Comment:
a better name can be used since any function that receives arguments is considered passing a variable. Could you please try to capture what the function actually does? Ex) `sync_with_main_branch`